fuse/ipns/ipns_unix.go

```diff
@@ -309,279 +271,130 @@ func (s *Node) ReadDirAll(ctx context.Context) ([]fuse.Dirent, error) {
-	if IpnsReadonly {
-		return nil
-	}
-
-	// If a write has happened
-	if n.dagMod != nil {
-		newNode, err := n.dagMod.GetNode()
-		if err != nil {
-			return err
-		}
-
-		if n.parent != nil {
-			log.Error("updating self in parent!")
-			err := n.parent.update(n.name, newNode)
-			if err != nil {
-				log.Criticalf("error in updating ipns dag tree: %s", err)
-				// return fuse.ETHISISPRETTYBAD
-				return err
-			}
-		}
-		n.Nd = newNode
-
-		/*/TEMP
-		dr, err := mdag.NewDagReader(n.Nd, n.Ipfs.DAG)
-		if err != nil {
-			log.Critical("Verification read failed.")
-		}
-		b, err := ioutil.ReadAll(dr)
-		if err != nil {
-			log.Critical("Verification read failed.")
-		}
-		fmt.Println("VERIFICATION READ")
-		fmt.Printf("READ %d BYTES\n", len(b))
-		fmt.Println(string(b))
-		fmt.Println(b)
-		//*/
-
-		n.dagMod = nil
-
-		n.wasChanged()
-	}
 	return nil
 }
 
-// Signal that a node in this tree was changed so the root can republish
-func (n *Node) wasChanged() {
-	if IpnsReadonly {
-		return
-	}
-	root := n.nsRoot
-	if root == nil {
-		root = n
-	}
-
-	root.repub.Publish <- struct{}{}
+func (fi *File) Flush(ctx context.Context, req *fuse.FlushRequest) error {
+	return fi.fi.Close()
 }
 
-func (n *Node) republishRoot() error {
-
-	// We should already be the root, this is just a sanity check
-	var root *Node
-	if n.nsRoot != nil {
-		root = n.nsRoot
-	} else {
-		root = n
-	}
+func (fi *File) Fsync(ctx context.Context, req *fuse.FsyncRequest) error {
+	return fi.fi.Flush()
+}
 
-	// Add any nodes that may be new to the DAG service
-	err := n.Ipfs.DAG.AddRecursive(root.Nd)
+func (fi *File) Forget() {
+	err := fi.fi.Flush()
 	if err != nil {
-		log.Criticalf("ipns: Dag Add Error: %s", err)
-		return err
+		log.Debug("Forget file error: ", err)
 	}
+}
 
-	ndkey, err := root.Nd.Key()
+func (dir *Directory) Mkdir(ctx context.Context, req *fuse.MkdirRequest) (fs.Node, error) {
+	child, err := dir.dir.Mkdir(req.Name)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	err = n.Ipfs.Namesys.Publish(n.Ipfs.Context(), root.key, ndkey)
-	if err != nil {
-		return err
-	}
-	return nil
+	return &Directory{dir: child}, nil
 }
 
-func (n *Node) Fsync(ctx context.Context, req *fuse.FsyncRequest) error {
-	return nil
-}
-
-func (n *Node) Mkdir(ctx context.Context, req *fuse.MkdirRequest) (fs.Node, error) {
-	if IpnsReadonly {
-		return nil, fuse.EPERM
-	}
-	dagnd := &mdag.Node{Data: ft.FolderPBData()}
-	nnode := n.Nd.Copy()
-	nnode.AddNodeLink(req.Name, dagnd)
-
-	child := &Node{
-		Ipfs: n.Ipfs,
-		Nd:   dagnd,
-		name: req.Name,
-	}
-
-	if n.nsRoot == nil {
-		child.nsRoot = n
-	} else {
-		child.nsRoot = n.nsRoot
-	}
-
-	if n.parent != nil {
-		err := n.parent.update(n.name, nnode)
+func (fi *File) Open(ctx context.Context, req *fuse.OpenRequest, resp *fuse.OpenResponse) (fs.Handle, error) {
+	if req.Flags&fuse.OpenTruncate != 0 {
+		log.Warning("Need to truncate file!")
+		err := fi.fi.Truncate(0)
```

Comment:
shouldnt truncation happen on the open? Maybe I dont remember proper unix and am only basic on editor, but afair, open with truncate only truncates on writing? hmmm not sure.